crates/oxc_transformer/src/es2018/object_spread.rs

```diff
@@ -0,0 +1,179 @@
+use oxc_ast::ast::*;
+use oxc_semantic::{ReferenceFlags, SymbolFlags};
+use oxc_span::SPAN;
+use oxc_traverse::{Traverse, TraverseCtx};
+use serde::Deserialize;
+
+use crate::{
+    context::Ctx,
+    helpers::{bindings::BoundIdentifier, module_imports::NamedImport},
+};
+
+#[derive(Debug, Default, Clone, Copy, Deserialize)]
+#[serde(default, rename_all = "camelCase")]
+pub struct ObjectSpreadOptions {
+    #[serde(alias = "loose")]
+    set_spread_properties: bool,
+    use_built_ins: bool,
+}
+
+pub struct ObjectSpread<'a> {
```

Comment:
Babel also handles the transformation for `rest` expressions, but this feature is not included in this pull request. Moreover, there is a issue with Babel's handling of `rest` parameters; I recommend emulating the behavior of esbuild. See https://github.com/swc-project/swc/issues/8527

Reply:
Although doesn't support the `rest` expression yet, we still should use `ObjectRestSpread` as a plugin name. Because we need to transform both `rest` and `spread` in the same plugin. We should always follow `Babel` and [Proposal](https://github.com/tc39/proposal-object-rest-spread) as possible

> Moreover, there is a issue with Babel's handling of rest expressions; I recommend emulating the behavior of esbuild. See https://github.com/swc-project/swc/issues/8527

Is possible to fix the issue in Babel's implementation? We should follow Babel's implementation as much as possible for now. This way we can reuse `Babel's` tests and easier debugging of errors caused by porting wrong

Does this make sense to you?

Reply:
Done.